crypto/threads_none.c

```diff
@@ -25,19 +25,22 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
 
 int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 {
-    OPENSSL_assert(*(unsigned int *)lock == 1);
+    if (!ossl_assert(*(unsigned int *)lock == 1))
+        return 0;
```

Comment:
> illustration to my argument

I mean I feel that we haven't come to consensus. Or have we? I mean some say "no crashes in library under no circumstances whatsoever [even if we reliably detect memory corruption (like in memory allocation, glibc sucks, etc.)], some are not as categorical.